src/Microsoft.Dnx.Runtime/PathSearchBasedProjectResolver.cs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Dnx.Runtime
+{
+    public class PathSearchBasedProjectResolver : IProjectResolver
```

Comment:
Because project name is not determined by folder name alone. Project needs to be load in to parse the name. It can't be indexed when project.json files are collected.